presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -700,7 +701,12 @@ protected Type visitTimeLiteral(TimeLiteral node, StackableAstVisitorContext<Con
         protected Type visitTimestampLiteral(TimestampLiteral node, StackableAstVisitorContext<Context> context)
         {
             try {
-                parseTimestampLiteral(session.getTimeZoneKey(), node.getValue());
+                if (SystemSessionProperties.isLegacyTimestamp(session)) {
+                    parseTimestampLiteral(session.getTimeZoneKey(), node.getValue());
```

Comment:
it would be cleaner if the method would be named: `parseLegacyTimestampLiteral`. Here and in other such `if`s below.